### DISCUSSION TITLE
Limit choices in headingsPlugin()

### DISCUSSION THREAD

**Discussion by sp88011** [mdx-editor/editor#52] [Q&A] [answered]
I love mdx-editor! Great work. 

I'm wondering how I would go about customizing which choices the headingsPlugin provides? 

For example, how do I only allow up to three heading levels?

<img width="204" alt="image" src="https://github.com/mdx-editor/editor/assets/5287078/7eaebb77-380d-46fe-a82e-5b80f657ebb6">

Is there any documentation on the plugins? Sorry if I missed it...

**Comment by petyosi** [comment, marked as answer]
That's a great practical suggestion - thank you very much. 

I just published v0.17.0. Now, the `headingsPlugin` accepts `allowedHeadingLevels` configuration parameter, which lets you limit the headings allowed. This parameter affects the `BlockTypeSelect` component and the markdown / google docs-like keyboard shortcuts. Of course, if you allow source mode, the user will still be able to insert a non-allowed heading directly, but I assume this is not what you're after anyway. 

Please let me know if the above works for you.
